doc/sphinx/language/extensions/evars.rst

```diff
@@ -74,18 +74,51 @@ Expressions often contain redundant pieces of information. Subterms that can be
 automatically inferred by Coq can be replaced by the symbol ``_`` and Coq will
 guess the missing piece of information.
 
+Automatic resolution of existential variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some tactics automatically resolve goals that are created when existential
+variables are created.
+
+.. _automatic-evar-resolution:
+
+.. example:: Automatic resolution of existential variables
+
+   .. coqtop:: reset in
+
+      Goal forall (a : nat), a = a -> exists n : nat, n = n.
+      Proof.
+        intros a e.
+
+   .. coqtop:: all
+
+       eexists ?[x].  (* creates a shelved goal *)
+       Unshelve.
+       exact e.  (* resolves the second goal automatically *)
+
+   Note the second goal would be resolved even without the :n:`Unshelve.`
+
+   The second goal (:n:`nat`) asks for a proof that :n:`nat` has a
+   :term:`witness`, i.e. it is not an empty type.  It can be proved directly
+   by applying a constructor of :n:`nat`, for example:
+
+   .. coqtop:: reset in
+
+      Goal nat.
+      apply O.  (* letter "O" is the constructor for the number "0" *)
```

Comment:
You could also write `0%nat` if I am not mistaken.

Reply:
Or just `0` or `1` ...